src/System.Private.CoreLib/shared/System/Environment.WinRT.cs

```diff
@@ -3,14 +3,22 @@
 // See the LICENSE file in the project root for more information.
 
 using System.IO;
+using Internal.Runtime.Augments;
 
 namespace System
 {
     public static partial class Environment
     {
         public static string UserName => "Windows User";
+
         public static string UserDomainName => "Windows Domain";
-        private static string GetFolderPathCore(SpecialFolder folder, SpecialFolderOption option) =>
-            WinRTFolderPaths.GetFolderPath(folder, option);
+
+        private static string GetFolderPathCore(SpecialFolder folder, SpecialFolderOption option)
+        {
+            WinRTInteropCallbacks callbacks = WinRTInterop.UnsafeCallbacks;
```

Comment:
@MichalStrehovsky, can I ask for your help here (not yet)?

Currently Environment is split between the runtime and corefx, with some of the implementation defined in an Environment in corefx, and the rest defined in the runtime, with an internal Environment and a public EnvironmentAugments that the public Environment calls to.

We're removing all of that cruft by just moving Environment to be entirely defined in the runtime, but that causes a complication here for Environment.GetFolderPath, as it depends on a WinRT API.  I've put all of that logic into a static WinRTFolderPaths.GetFolderPath method, which lives in the shared folder, and it's what corefx has been using.  But to utilize that from .NET Native/corert, it'll need to live behind this WinRTInteropCallbacks type, and while I've added a GetFolderPath abstract method to the base, the derived type that contains all of the actual logic isn't in the corert repo.

As part of your mirroring this over for .NET Native, would you be able to add the appropriate override to that type to call WinRTFolderPaths.GetFolderPath?  Hopefully it'll be that simple.

This PR as it stands isn't mergeable... there's a change going through right now in coreclr, and then when the mirror copies the changes files over to corert, we'll need to cherry-pick this change into that.

Reply:
Yup, I'll look into it once this merges. If I need help, I'll ask but it should be straightforward.

The bigger problem is that the closed source side is stuck with .NET Core 2.1 version of CoreFX for the time being (so we are going to end up with two copies of System.Environment). I'm hoping that I can revive a piece of the .NET Native compiler that we used to solve problems like this when we were bringing NetStandard 2.0 up and type moves like this were on daily schedule.

Reply:
Thanks.